source/core_mqtt.c

```diff
@@ -628,20 +640,30 @@ static int32_t sendPacket( MQTTContext_t * pContext,
             bytesRemaining -= ( size_t ) bytesSent;
             totalBytesSent += bytesSent;
             pIndex += bytesSent;
-            LogDebug( ( "BytesSent=%ld, BytesRemaining=%lu,"
-                        " TotalBytesSent=%ld.",
+            LogDebug( ( "BytesSent=%ld, BytesRemaining=%lu",
                         ( long int ) bytesSent,
-                        ( unsigned long ) bytesRemaining,
-                        ( long int ) totalBytesSent ) );
+                        ( unsigned long ) bytesRemaining ) );
+        }
+        else
+        {
+            /* No bytes were sent over the network. */
+            timeSinceLastSendMs = calculateElapsedTime( pContext->getTime(), lastSendTimeMs );
+
+            /* Check for timeout if we have been waiting to send any data over the network. */
+            if( timeSinceLastSendMs >= MQTT_SEND_RETRY_TIMEOUT_MS )
+            {
+                LogError( ( "Unable to send packet: Timed out in sending any data through transport send." ) );
```

Comment:
It sounds like it uses "send" one too many times
```suggestion
                LogError( ( "Unable to send packet: Timed out in transport send." ) );
```